### PR TITLE
Correct LearningToPaint to match the wild

### PR DESCRIPTION
**Pull request by aaronenyeshi**
Previously this model was only running the Stroked-Based
Renderer model, which is a simple nn with linear and conv
layers. But the real LearningToPaint model consists of
WGAN, discriminator and renderer. Adding the correct model
implementation.

### CODE REVIEW

**Comment by aaronenyeshi** [comment]
Here are the collected traces for updated PR:
Train:
image

Eval:
image

**Comment by xuzhao9** [comment]
Thanks for improving the LearningToPaint model quality. One question I have is do we observe similar GPU utilization pattern in the original LearningToPaint paper code demo? It is the ground truth and we would like to confirm that our reproduction matches the ground truth.

File: torchbenchmark/models/LearningToPaint/__init__.py [hunk 1]
```diff
@@ -1,99 +1,99 @@
...
 from torchbenchmark.tasks import REINFORCEMENT_LEARNING
 
 from argparse import Namespace
 
 torch.manual_seed(1337)
 np.random.seed(1337)
-torch.backends.cudnn.deterministic = True
-torch.backends.cudnn.benchmark = False
+torch.backends.cudnn.deterministic = False
+torch.backends.cudnn.benchmark = True
```
Comment (xuzhao9) [inline torchbenchmark/models/LearningToPaint/__init__.py:18]: Maybe we should set this config globally instead of setting it for individual models repeatedly. But we can do it in a separate PR.

File: torchbenchmark/models/LearningToPaint/__init__.py [hunk 2]
```diff
@@ -1,99 +1,99 @@
...
 
     def get_module(self):
-        return self.module,[self.example_inputs]
-
-    # LearningToPaint has another model
-    # instance for inference that has
-    # already been optimized for inference
-    def set_eval(self):
-        pass
+        raise NotImplementedError()
```
Comment (xuzhao9) [inline torchbenchmark/models/LearningToPaint/__init__.py:54]: This API has some users (e.g., https://github.com/pytorch/benchmark/pull/479), can you implement this (e.g. using randomly generate inputs to the actor model)?

**Comment by aaronenyeshi** [comment]
> Thanks for improving the LearningToPaint model quality. One question I have is do we observe similar GPU utilization pattern in the original LearningToPaint paper code demo? It is the ground truth and we would like to confirm that our reproduction matches the ground truth to help confirm that the bad utilization is a problem of user code, not our reproduction.

Here is a 1 second snippet of the eval trace in the original LearningToPaint code:
image

File: torchbenchmark/models/LearningToPaint/__init__.py (hunk 1 above)
Comment (aaronenyeshi) [reply torchbenchmark/models/LearningToPaint/__init__.py:18]: Sounds good to me!

File: torchbenchmark/models/LearningToPaint/__init__.py (hunk 2 above)
Comment (aaronenyeshi) [reply torchbenchmark/models/LearningToPaint/__init__.py:54]: Sure, I can take a further look into this.

Comment (aaronenyeshi) [reply torchbenchmark/models/LearningToPaint/__init__.py]: done!

**Comment by aaronenyeshi** [comment]
This implementation uses the parameters from LearningToPaint Source code and research paper.

**Review by xuzhao9 — APPROVED** [review]
LGTM. Thanks for improving this model!
